public/view/overlay-controls/admin-search.jsx

```diff
@@ -0,0 +1,52 @@
+var React = require('react');
+var Typeahead = require('react-typeahead').Typeahead;
+require('./admin-search.css');
+
+var AdminSearch = React.createClass({
+  admins: function() {
+    var admin_objs = this.props.admin_details.admin_data_by_code;
+
+    // Maintain array of selected country iso codes
+    var available_countries = Object.keys(
+      this.props.selected_countries.selected_country_codes
+    );
+
+    // Maintain list of admins that belong to selected countries
+    var admins = Object.keys(this.props.admin_details.admin_data_by_code)
+    .filter(function(key) {
+      var admin_code = admin_objs[key].admin_code.split('-')[0];
+      return available_countries.find(function(e) {
+        return e === admin_code;
+      });
+    // Return array of admin names
+    }).map(function(key) {
+      return admin_objs[key].name + ' ' +
+      admin_objs[key].admin_code;
+    }).sort();
+    return admins;
+  },
+
+  optionSelected: function() {
+    var argument = arguments[0];
+    var admin_objs = this.props.admin_details.admin_data_by_code;
+    var admin_code = Object.keys(this.props.admin_details.admin_data_by_code)
+    .find(function(key) {
+      return admin_objs[key].name + ' ' + admin_objs[key].admin_code === argument;
```

Comment:
up to you: you could build up a map from search key to admin code, on line 23 above, and just do a lookup here. That way, if you change how something is updated, you can't update one call site but not the other. It's not a big deal though.